### PR TITLE
Finalize autoMarket

### PR DESCRIPTION
**Pull request by dlaxcess**
- [x] sell modal
- [x] buy modal
- [x] curency from chainId

### CODE REVIEW

File: common/lib/nft/kautomarket.ts [hunk 1]
```diff
@@ -196,7 +196,7 @@ async function* setTokenPrice(
 
   if (!collection.supports?.IOpenMarketable) return;
 
-  const txResp = await (contract as IOpenMarketable).setTokenPrice(BigNumber.from(tokenID), tokenPrice);
+  const txResp = await (contract as IOpenMarketable).setTokenPrice(BigNumber.from(tokenID), BigNumber.from(tokenPrice));
```
Comment (zapaz) [inline common/lib/nft/kautomarket.ts:199]: tokenPrice should be BigNUmber => change param type

File: svelte/components/Collection/CollectionCreate.svelte [hunk 2]
```diff
@@ -140,8 +139,8 @@
       template,
       $metamaskSigner,
       ethers.utils.parseEther(inputPrice) || BigNumber.from(0),
-      inputReceiver,
-      BigNumber.from(Math.round(Number(inputFee) * 100))
+      inputReceiver || constants.AddressZero,
+      BigNumber.from(Math.round(Number(inputFee) * 100)) || BigNumber.from(0)
```
Comment (zapaz) [inline svelte/components/Collection/CollectionCreate.svelte:143]: shorter :  BigNumber.from((Number(inputFee)||0) * 100)

File: svelte/components/Collection/CollectionCreate.svelte [hunk 3]
```diff
@@ -230,8 +229,8 @@
           </div>
 
           {#if template === "OpenAutoMarket/ownable"}
-            <div class="section">
```
Comment (zapaz) [inline svelte/components/Collection/CollectionCreate.svelte:233]: should be display if price > 0 and not collection owner

File: svelte/components/Nft/Nft.svelte [hunk 4]
```diff
@@ -80,9 +83,9 @@
 
         {#if $nft.collection?.supports?.IOpenMarketable}
           {#if $nft.owner === account}
-            <NftSell {chainId} {address} {tokenID} nftPrice={String($nft?.price) || "0"} />
+            <NftSell {chainId} {address} {tokenID} nft={$nft} />
```
Comment (zapaz) [inline svelte/components/Nft/Nft.svelte:86]: use chainId/address/tokenID  or nft to identify NFT not both

File: svelte/components/Nft/Nft.svelte [hunk 5]
```diff
@@ -145,7 +148,8 @@
                 <div class="flex"><span class="label">Nft Price</span></div>
                 <div class="flex">
                   <span class="overflow-ellipsis" title={ethers.utils.formatEther($nft.price)} target="_blank">
-                    {ethers.utils.formatEther($nft.price)} Eth
+                    {ethers.utils.formatEther($nft.price)}
```
Comment (zapaz) [inline svelte/components/Nft/Nft.svelte:151]: import {utils} from "ethers"    then use utilsformatEther

File: svelte/components/Nft/Nft.svelte [hunk 6]
```diff
@@ -155,10 +159,16 @@
...
-                    <span class="overflow-ellipsis" title={String($nft.royaltyFee)}>No royalties amount setted</span>
+                    <span class="overflow-ellipsis" title={String($nft.royaltyFee.toNumber() / 100)}
+                      >No royalties amount setted</span
+                    >
                   {:else}
-                    <span class="link overflow-ellipsis" title={`${$nft.royaltyFee.div(100)} %`} target="_blank">
-                      {$nft.royaltyFee.div(100)} %
+                    <span
+                      class="link overflow-ellipsis"
+                      title={`${$nft.royaltyFee.toNumber() / 100} %`}
```
Comment (zapaz) [inline svelte/components/Nft/Nft.svelte:168]: use Number ( var ) or var.toNumber()   not both

rather Number(var)

File: svelte/components/Nft/Nft.svelte [hunk 7]
```diff
@@ -169,33 +179,35 @@
...
-                      ? "No receiver setted for Royalties"
-                      : $nft.royaltyAccount}
+                    {#if $nft.royaltyAccount === constants.AddressZero}
+                      "No receiver setted for Royalties"
+                    {:else}
+                      {@html explorerAddressLink(chainId, $nft.royaltyAccount, 15)}
+                    {/if}
                   </span>
-                  {#if $nft.owner === account && $nft.collection?.supports?.IOpenMarketable && $nft.royaltyFee.eq(0) && $nft.royaltyAccount === constants.AddressZero}
+                  <!-- {#if $nft.owner === account && $nft.collection?.supports?.IOpenMarketable && $nft.royaltyAmount === "0" && $nft.royaltyReceiver === constants.AddressZero}
```
Comment (zapaz) [inline svelte/components/Nft/Nft.svelte:188]: it's BigNumber =>  $nft.royaltyAmount.eq(0)

File: svelte/components/Nft/NftBuy.svelte [hunk 8]
```diff
@@ -72,7 +84,7 @@
   });
 
   const buyConfirm = async () => {
-    const buyTxRespYield = buyNft(chainId, address, tokenID, nftPrice, $metamaskSigner);
+    const buyTxRespYield = buyNft(chainId, address, tokenID, nft.price.toString(), $metamaskSigner);
```
Comment (zapaz) [inline svelte/components/Nft/NftBuy.svelte:87]: use BigNumber   => nft.price simply

File: svelte/components/Nft/NftBuy.svelte [hunk 9]
```diff
@@ -95,11 +107,11 @@
 
 <a
   on:click={() => {
-    if (nftPrice !== "0") open = true;
+    if (!nft.price.isZero()) open = true;
```
Comment (zapaz) [inline svelte/components/Nft/NftBuy.svelte:110]: nft.price.eq(0)

File: svelte/components/Nft/NftBuy.svelte [hunk 10]
```diff
@@ -95,11 +107,11 @@
...
   }}
   href="#buy-nft-{tokenID}"
-  class="btn-buy-modal {nftPrice === '0' ? 'kre-disabled' : ''}"
-  title="Buy this nft"><i class="fa fa-shopping-cart fa-left" aria-disabled={nftPrice === "0"} /> Buy</a
+  class="btn-buy-modal {nft.price.isZero() ? 'kre-disabled' : ''}"
```
Comment (zapaz) [inline svelte/components/Nft/NftBuy.svelte:113]: idem

File: svelte/components/Nft/NftBuy.svelte (hunk 8 above)
Comment (zapaz) [reply svelte/components/Nft/NftBuy.svelte:87]: change lib kbuy.ts accordingly

File: svelte/components/Nft/NftMint.svelte [hunk 11]
```diff
@@ -318,22 +321,30 @@
                     <div>
                       <span class="kre-market-info-title label-big kre-no-wrap-title">Mint price</span>
                       <span class="kre-market-info-value label-big kre-no-wrap-title"
-                        >{utils.formatEther(collection?.price)} (Eth)</span
+                        >{utils.formatEther(collection?.price)} ({getNetwork(chainId).nativeCurrency.symbol})</span
```
Comment (zapaz) [inline svelte/components/Nft/NftMint.svelte:324]: create a lib function named   getCurrency(chainId)

File: svelte/components/Nft/NftMint.svelte [hunk 12]
```diff
@@ -318,22 +321,30 @@
...
                       >
                     </div>
                   {/if}
                   {#if collection?.royaltyFee?.gt(0)}
                     <div>
                       <span class="kre-market-info-title label-big">royalties</span>
-                      <span class="kre-market-info-value label-big">{collection?.royaltyFee.div(100)} %</span>
+                      <span class="kre-market-info-value label-big">{collection?.royaltyFee.toNumber() / 100} %</span>
```
Comment (zapaz) [inline svelte/components/Nft/NftMint.svelte:331]: no BigNumber div

File: svelte/components/Nft/NftMint.svelte [hunk 13]
```diff
@@ -318,22 +321,30 @@
...
+                      <span class="kre-market-info-value label-big overflow-ellipsis"
+                        ><a href={explorerAddressUrl(chainId, collection?.royaltyAccount)} class="link"
+                          >{collection?.royaltyAccount}</a
+                        ></span
+                      >
                     </div>
                   {/if}
+                  <div>
+                    <span class="kre-market-info-title label-big kre-no-wrap-title">fees</span>
+                    <span class="kre-market-info-value label-big overflow-ellipsis">{config.treasury.fee / 100} %</span>
```
Comment (zapaz) [inline svelte/components/Nft/NftMint.svelte:346]: use BigNumber (or Number ok for fees %)

File: svelte/components/Nft/NftSetPrice.svelte [hunk 14]
```diff
@@ -18,9 +22,11 @@
   export let chainId: number;
   export let address: string;
   export let tokenID: string;
-  export let nftPrice: string;
+  export let nft: NftType;
```
Comment (zapaz) [inline svelte/components/Nft/NftSetPrice.svelte:25]: not both nft and chainID...

File: svelte/components/Nft/NftSetPrice.svelte [hunk 15]
```diff
@@ -65,14 +71,34 @@
...
     tokenSetPriceInit();
   });
 
   const setPriceConfirm = async () => {
-    const tokenSetPriceTxRespYield = setTokenPrice(chainId, address, tokenID, tokenNewPrice, $metamaskSigner);
+    const tokenSetPriceTxRespYield = setTokenPrice(
+      chainId,
+      address,
+      tokenID,
+      utils.parseEther(tokenNewPrice).toString(),
```
Comment (zapaz) [inline svelte/components/Nft/NftSetPrice.svelte:99]: use BigNumber

File: svelte/components/Nft/NftSetPrice.svelte [hunk 16]
```diff
@@ -90,28 +116,46 @@
...
+    collectionStore.refreshOne(chainId, address, $metamaskAccount).catch(console.error);
+    await nftStore.refreshSubList(chainId, address, $metamaskAccount);
     await nftStore.refreshOne(chainId, address, tokenID).catch(console.error);
   };
 </script>
 
 {#if tokenPriceSetting == S1_CONFIRM}
   <div class="titre">
-    <p><i class="fas fa-angle-right" /> Set this NFT #{tokenID} price</p>
+    <p><i class="fas fa-angle-right" /> List item #{tokenID} for sale using Kredeum AutoMarket smartcontract</p>
```
Comment (zapaz) [inline svelte/components/Nft/NftSetPrice.svelte:132]: using AUtomarket smartcontract will suffice

**Review by zapaz — COMMENTED** [review]
lot of remarks but quite small ones

File: svelte/components/Collection/CollectionCreate.svelte (hunk 3 above)
Comment (dlaxcess) [reply svelte/components/Collection/CollectionCreate.svelte:233]: collection is not created so price and owner are undefined ?

Comment (zapaz) [reply svelte/components/Collection/CollectionCreate.svelte:233]: ok, so to display with no condition ?

File: svelte/components/Nft/NftMint.svelte (hunk 13 above)
Comment (dlaxcess) [reply svelte/components/Nft/NftMint.svelte:346]: config.treasury.fee is Number ?

Comment (zapaz) [reply svelte/components/Nft/NftMint.svelte:361]: yes read value comes from json, so Number

**Comment by zapaz** [comment]
directly merged into integ

**Pull request closed without merging**